### DISCUSSION TITLE
fail logic

### DISCUSSION THREAD

**Discussion by mexoboy** [deployphp/deployer#2655] [Help needed]
Hi everyone. 
I want to integrate smoke tests after deploy. For that I make custom task and add it after deploy:symlink. If any test failed, release must do rollback. I have recipe something like this:
`
task('deploy', [
   ...
   'deploy:symlink',
   'deploy:test',
   'deploy:unlock',
    'cleanup',
]);

after('deploy', 'success');
after('deploy:failed', 'deploy:unlock');
fail('deploy:test', 'rollback');
`
The problem: When some test failed, inside task "deploy:test" my code throwing Deployer\Exception\Exception. I expected, than after failure triggered task "rollback". But it doesn't happen. 

Anyone can explain me what I do wrong? Thanks for answers and advices.

**Comment by zahhar** [comment]
Put in this way: 
```
after('deploy:failed', 'rollback');
```

Your rollback task must include unlock and everything else you need.
